### PR TITLE
Removing Octokit

### PR DESCRIPTION
**Pull request by StanleyGoldman**
This targets #599 and does the work of removing Octokit from the application.
This includes libraries and a special solution used for debugging Octokit.
The majority of code changes to the application are removing `IGitHubClient` and credentials objects.

### CODE REVIEW

File: src/GitHub.Api/Application/ApiClient.cs [hunk 1]
```diff
@@ -214,12 +211,12 @@ public ApiClient(UriString hostUrl, IKeychain keychain, IGitHubClient githubClie
...
                 var command = new StringBuilder("publish -r ");
-                command.Append(newRepository.Name);
+                command.Append(repositoryName);
 
-                if (!string.IsNullOrEmpty(newRepository.Description))
+                if (!string.IsNullOrEmpty(description))
                 {
                     command.Append(" -d ");
-                    command.Append(newRepository.Description);
+                    command.Append(description);
```
Comment (shana) [inline src/GitHub.Api/Application/ApiClient.cs]: This needs quotes around it for the command line to parse properly

File: src/GitHub.Api/Application/ApplicationConfiguration.cs [hunk 2]
```diff
@@ -11,19 +10,13 @@ static ApplicationConfiguration()
         {
             var executingAssembly = typeof(ApplicationConfiguration).Assembly;
             AssemblyName = executingAssembly.GetName();
-            ProductHeader = new ProductHeaderValue(ApplicationInfo.ApplicationSafeName, AssemblyName.Version.ToString());
```
Comment (shana) [inline src/GitHub.Api/Application/ApplicationConfiguration.cs]: Not related directly to this PR, but you should make sure this information is sent along with the requests in the node app, so you might need to set it in the environment as well.

File: src/GitHub.Api/Authentication/LoginManager.cs [hunk 3]
```diff
@@ -83,108 +73,59 @@ class LoginManager : ILoginManager
...
-                {
-                    auth = new ApplicationAuthorization(password);
-                }
-                else
-                {
-                    await keychain.Clear(host, false);
-                    return new LoginResultData(LoginResultCodes.Failed, Localization.LoginFailed, host);
-                }
+                await keychain.Clear(host, false);
+                return new LoginResultData(LoginResultCodes.Failed, Localization.LoginFailed, host);
```
Comment (shana) [inline src/GitHub.Api/Authentication/LoginManager.cs]: This is missing the other error conditions like locked out and the workaround for enterprise, isn't it?

File: src/GitHub.Api/Application/ApiClient.cs (hunk 1 above)
Comment (StanleyGoldman) [reply src/GitHub.Api/Application/ApiClient.cs]: Addressed in #599

File: src/GitHub.Api/Application/ApplicationConfiguration.cs (hunk 2 above)
Comment (StanleyGoldman) [reply src/GitHub.Api/Application/ApplicationConfiguration.cs]: Tracked in #617

File: src/GitHub.Api/Authentication/LoginManager.cs (hunk 3 above)
Comment (StanleyGoldman) [reply src/GitHub.Api/Authentication/LoginManager.cs]: Tracked in #617

File: src/GitHub.Api/Application/ApplicationConfiguration.cs (hunk 2 above)
Comment (shana) [reply src/GitHub.Api/Application/ApplicationConfiguration.cs]: This should be fixed in #599

**Pull request merged**